postmaster/static/js/admins.js

```diff
@@ -190,6 +222,39 @@ function adminEventListeners () {
 
     $('#filterRow input').typeWatch(typeWatchOptions);
 
+    twoFAModal.on('show.bs.modal', function (e) {
+      var id = $(e.relatedTarget).data('pk');
+      $.ajax({
+        url: '/api/v1/admins/' + id + '/2factor/qrcode',
+        success: function(data){
+           $("#qrcode").html(new XMLSerializer().serializeToString(data.documentElement));
+           verify2FABtn.attr('data-pk', id);
+        },
+        error: function (response) {
+          addStatusMessage('error', filterText(jQuery.parseJSON(response.responseText).message));
+          twoFAModal.modal('hide');
```

Comment:
This is very jarring where the modal is shown for a split second and then disappears with an error. The modal should only appear on success of loading the QR code